### PR TITLE
DispatcherQueue updates for WinUI3

### PR DESCRIPTION
**Pull request by MikeHillberg**

### CODE REVIEW

File: winui3/DispatcherQueueUpdates.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+# Examples
+
+This example shows a thread ensuring it has a DispatcherQueue, setting a Synchronization
+context onto the thread that uses that queue, and then goes into a message pump.
+
+```cs
+static void Run()
+{
+    // Ensure we have a DispatcherQueue on this thread.
```
Comment (jeffstall) [inline winui3/DispatcherQueueUpdates.md:83]: Please update comment:
// Note - this pattern is only necessary if you need to create a DispatcherQueue on a thread (ex as part
// of initialization).  For the vast majority of scenarios, a hosting framework has already created one
// on your behalf and so you should simply call DispatcherQueue.GetForCurrentThread() to retrieve it.

**Comment by MikeHillberg** [comment]
@scottflute - you're right. I removed that part of the proposal and will open a separate Issue on it.

File: winui3/DispatcherQueueUpdates.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+> In a WinUI you typically don't need to write this code for the initial UI thread, as it's
+automatically generated.
+
+```cs
+static void Run()
+{
+    // Ensure we have a DispatcherQueue on this thread. 
+    if(DispatcherQueue.GetForCurrentThread() == null)
+    {
+        DispatcherQueueController.CreateOnCurrentThread();
```
Comment (oldnewthing) [inline winui3/DispatcherQueueUpdates.md:125]: Does anything interesting happen when the controller is GC'd? Or is it fine to just throw it away - the thread retains a reference to its own controller, but that reference is inaccessilble so you have no way to shut down the dispatcher queue. Your only choice is to rip the rug out from under it at thread termination.

File: winui3/DispatcherQueueUpdates.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    }
+
+    var syncContext = new DispatcherQueueSyncronizationContext();
```
Comment (david-risney) [inline winui3/DispatcherQueueUpdates.md:128]: Typo: DispatcherQueueSync**h**ronizationContext

File: winui3/DispatcherQueueUpdates.md (hunk 1 above)
Comment (MikeHillberg) [reply winui3/DispatcherQueueUpdates.md:122]: @scott-moore-ms - look OK?

File: winui3/DispatcherQueueUpdates.md (hunk 2 above)
Comment (MikeHillberg) [reply winui3/DispatcherQueueUpdates.md:125]: @scott-moore-ms

Comment (scott-moore-ms) [reply winui3/DispatcherQueueUpdates.md:125]: It should be fine to throw the controller away.  

By doing that, as you say, you lose the ability to fetch it again, but that is intentional in the design of splitting the controller from the queue - the entity that created the controller is the only one that has the ability to shut down the queue, whereas anyone on a thread should be able to access the queue to schedule their own work.  If after creating the controller, you decide you don't need to explicitly shut it down in the future (which is fairly common), then discarding it is totally fine.

File: winui3/DispatcherQueueUpdates.md (hunk 1 above)
Comment (scott-moore-ms) [reply winui3/DispatcherQueueUpdates.md:122]: LGTM

**Review by scott-moore-ms — APPROVED** [review]

**Review by jeffstall — COMMENTED** [review]
1. We may need to change the namespace from Microsoft.System if we move DispatcherQueue.
2. It is still preferable to not have the "if not created, create it now" pattern. It's not good if components just start creating DispatcherQueues. This should almost always be created by the top-level object that created the thread, and it should know if the thread should have a DispatcherQueue.

**Review by jeffstall — APPROVED** [review]
(Marking approved to submit... we can continue to refine...)

**Pull request merged**